src/filtering/update.jl

```diff
@@ -193,6 +193,50 @@ function update!(
     return x_out, loglikelihood
 end
 
+function update!(
+    x_out::SRGaussian{T,<:MFBD},
+    x_pred::SRGaussian{T,<:MFBD},
+    measurement::Gaussian{
+        <:AbstractVector,
+        <:Union{<:PSDMatrix{T,<:MFBD},<:MFBD},
+    },
+    H::MFBD,
+    K1_cache::MFBD,
+    K2_cache::MFBD,
+    M_cache::MFBD,
+    C_dxd::MFBD,
+    C_d::AbstractVector;
+    R::Union{Nothing,PSDMatrix{T,<:MFBD}}=nothing,
+) where {T}
+    d = length(blocks(x_out.Σ.R))
+    q = size(blocks(x_out.Σ.R)[1], 1) - 1
+
+    ll = zero(eltype(x_out.μ))
+    for i in eachindex(blocks(x_out.Σ.R))
+        _, _ll = update!(
+            Gaussian(view(x_out.μ, (i-1)*(q+1)+1:i*(q+1)),
+                PSDMatrix(x_out.Σ.R.blocks[i])),
+            Gaussian(view(x_pred.μ, (i-1)*(q+1)+1:i*(q+1)),
+                PSDMatrix(x_pred.Σ.R.blocks[i])),
+            Gaussian(view(measurement.μ, i:i),
+                if measurement.Σ isa PSDMatrix
+                    PSDMatrix(measurement.Σ.R.blocks[i])
+                else
+                    measurement.Σ.blocks[i]
+                end),
+            H.blocks[i],
+            K1_cache.blocks[i],
+            K2_cache.blocks[i],
+            M_cache.blocks[i],
+            C_dxd.blocks[i],
+            view(C_d, i:i);
+            R,
+        )
+        ll += _ll
+    end
+    return x_out, ll
```

Comment:
Check that the loglikelihood that comes out is correct! To do this, compare two update steps, one with dense matrices and ne with Blockdiagonal matrices.

Reply:
Actually do the same for the IsometricKronecker case too!